modules/twinkleblock.js

```diff
@@ -950,6 +950,15 @@ Twinkle.block.blockPresetsInfo = {
 		reason: '{{uw-ublock-double}} <!-- Username closely resembles another user, hard block -->',
 		summary: 'You have been indefinitely blocked from editing because your [[WP:U|username]] appears to impersonate another established Wikipedia user'
 	},
+	'uw-upeblock': {
+		autoblock: true,
+		expiry: 'infinity',
+		forRegisteredOnly: true,
+		nocreate: true,
+		pageParam: true,
+		reason: '[[WP:PAID|Undisclosed paid editing]] in violation of the WMF [[WP:TOU|Terms of Use]]',
+		summary: 'You have been indefinitely blocked from editing because your account is being used only for [[WP:SPAM|spam or advertising]] in violation of [[WP:PAID|Wikipedia policy on paid advocacy'
```

Comment:
Missing brackets to close out the link:

```suggestion
		summary: 'You have been indefinitely blocked from editing because your account is being used only for [[WP:SPAM|spam or advertising]] in violation of [[WP:PAID|Wikipedia policy on paid advocacy]]'
```

Reply:
That being said, I think it'd be good to have this be in closer alignment with the log entry, if only by specifically stating it's for UPE.  Maybe something like:

```suggestion
		summary: 'You have been indefinitely blocked from editing because your account is being used in violation of [[WP:PAID|Wikipedia policy on undisclosed paid advocacy]]'
```

Or something like that?